examples/with-apollo-neo4j-graphql/apollo/type-defs.ts

```diff
@@ -0,0 +1,17 @@
+import { gql } from '@apollo/client'
```

Comment:
Do I understand properly that this is a file used only on the server? Then I don't think we need this dependency on `@apollo/client`. I'd `import gql from 'graphql-tag'` directly instead (this is just a re-export), or honestly just define the schema as a string (perhaps read with fs.readFileSync from a `schema.graphql` file for better editor support) rather than parsing it with the gql tag.

Reply:
@glasser It is also imported indirectly by the Apollo client code in much the same way the existing Apollo examples do it. 

Here's api-routes-apollo-server-and-client for example:
 
https://github.com/vercel/next.js/blob/ef2672d6c283507ca7ee0d8c767d187b5fb53bbd/examples/api-routes-apollo-server-and-client/apollo/client.tsx#L4-L11

I'm not 100% confident we need to cover the windowless case here actually, so we could potentially update the client.

As an aside, do you know if the client file needs the `.tsx` extension, or does `.ts` suffice in this case with the only React code being `useMemo`?